mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java

```diff
@@ -48,7 +48,7 @@
 
     @Override
     public boolean isAutomaticallySent(Mail mail) throws MessagingException {
-        return mail.getSender() == null ||
+        return !mail.hasSender() ||
```

Comment:
just a tiny thing, I see you use the negative boolean operation for `mail.hasSender()` not only once but many times. Could we have another method `mail.hasNoSender()` regarding to the readability?